docs/plugin_developers_guide/04_north_plugins.rst

```diff
@@ -16,578 +16,336 @@
-      "name": "Stats OMF to PI north",
-      "processName": "north_c",
-      "type": "INTERVAL",
-      "repeat": 30,
-      "time": 0,
-      "day": null,
-      "exclusive": true,
-      "enabled": true
+Python plugins are loaded dynamically and executed either within a task, known as the *sending_task* or *north* task. This code is implemented in C++ and embedded a Python interpreter that is used to run the Python plugin.
+
+The plugin_info call
+~~~~~~~~~~~~~~~~~~~~
+
+The *plugin_info* call is the first call that will be made to a plugin and is called only once. It is part of the standard plugin interface that is implemented by north, south, filter, notification rule and notification delivery plugins. No arguments are passed to this call and it should return a *plugin information structure* as a Python dict.
+
+A typical implementation for a simple north plugin simply returns a DICT as follows
+
+.. code-block:: python
+
+    def plugin_info():
+        return {
+            'name': 'http',
+            'version': '1.9.1',
+            'type': 'north',
+            'interface': '1.0',
+            'config': _DEFAULT_CONFIG
+        }
+
+In the case above *_DEFAULT_CONFIG* is another Python DICT that contains the defaults for the plugin configuration and will be covered in the next section.
+
+The items in the structure returned by *plugin_info* are
+
+.. list-table::
+    :header-rows: 1
+
+    * - Name
+      - Description
+    * - name
+      - The name of the plugin
+    * - version
+      - The version of the plugin. Typically this is the same as the version of Fledge it is designed to work with but is not constrained to be the same.
+    * - type
+      - The type of the plugin, in this case the type will always be *north*
+    * - interface
+      - The version of the plugin interface that the plugin supports. In this case the version if 1.0
+    * - config
+      - The DICT that defines the configuration that the plugin has as default.
+
+
+Configuration
+#############
+
+Configuration within Fledge is represented in a JSON structure that defines a name, value, default, type and a number of other parameters. The configuration process works by the plugins having a default configuration that they return from the plugin_init call. The Fledge configuration code will then combine this with a coy of that configuration that it holds. On the first time a service is created, with no previously held configuration, he configuration manager will take the default values and make those the actual values. The user may then update these to set non-default values. In subsequent executions of the plugin these values will be combined with the defaults to create the in use configuration that is passed to the *plugin_init* entry point. The mechanism is designed to allow initial execution of a plugin, but also to allow upgrade of a plugin to create new configuration items for the plugins whilst preserving previous configuration values set by the user.
```

Comment:
> a number of other parameters

a number of other optional parameters

> coy

copy

> he configuration manager

the configuration manager